src/utils.ts

```diff
@@ -68,17 +68,19 @@ export const mergeDeep = (
   return target
 }
 
-export function isClassComponent(component: any) {
-  // TypeScript
-  return (
-    component.toString().includes('_super.apply(this, arguments) || this') ||
-    // native ES6
-    (typeof component === 'function' &&
-      /^\s*class\s+/.test(component.toString()))
-  )
+export function isClassComponent(component: VNodeTypes) {
```

Comment:
while we're here, let's clean up all three helpers and get rid of `any`

Current approach of `isClassComponent` is super-flaky and does not work if (for example) babel instruments code after TypeScript or you're running instanbul or Wallaby for code coverage (which instruments code) 

New approach relies on [implementation detail](https://github.com/vuejs/vue-class-component/blob/next/src/vue.ts#L194) of `vue-class-component` but feels more reliable, since it will not accept any class and if `vue-class-component` will change implementation in future tests will warn us

Feel free to discard this change if you don't feel it is correct

Reply:
This seems more reliable - we (Vue team) control the `vue-class-component` library, and core has some code that also relies on `__vccOpts` existing.